gprofiler/main.py

```diff
@@ -171,25 +171,28 @@ def _generate_output_files(
         logger.info(f"Saved collapsed stacks to {collapsed_path}")
 
         if self._flamegraph:
-            flamegraph_path = base_filename + ".html"
-            flamegraph_data = (
-                Path(resource_path("flamegraph/flamegraph_template.html"))
-                .read_text()
-                .replace(
-                    "{{{JSON_DATA}}}",
-                    run_process(
-                        [resource_path("burn"), "convert", "--type=folded", collapsed_path], suppress_log=True
-                    ).stdout.decode(),
+            if os.path.exists(resource_path("burn")):
+                flamegraph_path = base_filename + ".html"
+                flamegraph_data = (
+                    Path(resource_path("flamegraph/flamegraph_template.html"))
+                    .read_text()
+                    .replace(
+                        "{{{JSON_DATA}}}",
+                        run_process(
+                            [resource_path("burn"), "convert", "--type=folded", collapsed_path], suppress_log=True
+                        ).stdout.decode(),
+                    )
+                    .replace("{{{START_TIME}}}", start_ts)
+                    .replace("{{{END_TIME}}}", end_ts)
                 )
-                .replace("{{{START_TIME}}}", start_ts)
-                .replace("{{{END_TIME}}}", end_ts)
-            )
-            Path(flamegraph_path).write_text(flamegraph_data)
+                Path(flamegraph_path).write_text(flamegraph_data)
 
-            # point last_flamegraph.html at the new file; and possibly, delete the previous one.
-            self._update_last_output("last_flamegraph.html", flamegraph_path)
+                # point last_flamegraph.html at the new file; and possibly, delete the previous one.
+                self._update_last_output("last_flamegraph.html", flamegraph_path)
 
-            logger.info(f"Saved flamegraph to {flamegraph_path}")
+                logger.info(f"Saved flamegraph to {flamegraph_path}")
+            else:
+                logger.info("'burn' not available, not writing flamegraph!")
```

Comment:
Hmm not sure that this is ideal, how hard is it to build burn for ARM? I think that generating local flamegraphs is a core feature that we don't want to ever skip

Reply:
Not hard probably. I'll try

Reply:
Just didn't do it at first

Reply:
Built